tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -170,24 +238,30 @@ private void ConfigureTraceTransport(IConfigurationSource source, bool isWindows
                             // backwards compatibility for names used in the past
                             source?.GetInt32("DATADOG_TRACE_AGENT_PORT");
 
+            TracesPipeName = source?.GetString(ConfigurationKeys.TracesPipeName);
+
             // Agent port is set to zero in places like AAS where it's needed to prevent port conflict
-            // The agent will fail to start if it can not bind a port
-            var hasExplicitTcpConfig = agentPort != 0 && agentHost != null;
+            // The agent will fail to start if it can not bind a port, so we need to override 8126 to prevent port conflict
+            // Port 0 means it will pick some random available port
+            var hasExplicitHostOrPortSettings = (agentPort != null && agentPort != 0) || agentHost != null;
 
-            if (hasExplicitTcpConfig)
+            if (hasExplicitHostOrPortSettings)
             {
-                // If there is any explicit configuration for TCP, prioritize it
-                traceTransport = TracesTransportType.Default;
-            }
-            else if (isWindows)
-            {
-                // Check for explicit windows named pipe config
-                TracesPipeName = source?.GetString(ConfigurationKeys.TracesPipeName);
-
-                if (TracesPipeName != null)
+                if (agentHost?.StartsWith(UnixDomainSocketPrefix) ?? false)
                 {
-                    traceTransport = TracesTransportType.WindowsNamedPipe;
+                    traceTransport = TracesTransportType.UnixDomainSocket;
+                    TracesUnixDomainSocketPath = agentHost;
                 }
+                else
+                {
+                    // The agent host is explicitly configured, we should assume UDP for metrics
+                    forceMetricsOverUdp = true;
+                    traceTransport = TracesTransportType.Default;
+                }
+            }
+            else if (!string.IsNullOrWhiteSpace(TracesPipeName))
+            {
+                traceTransport = TracesTransportType.WindowsNamedPipe;
```

Comment:
Should we only be trying to enable WindowsNamedPipe if we're on Windows?

Reply:
I'd say we could do that, but if someone wants to explicitly configure named pipes, that's on them in my opinion.
I removed the `is windows` checks because I felt they added no real value.

Reply:
+1. If we're not on Windows and there's a WindowsNamedPipe variable, I think we should log a warning to say that the variable was detected but unused.

Reply:
They will already get a log that states they are using this transport, and will get connection errors when submitted. I'll add another log if that seems important, though I doubt it will ever be of use.